### PR TITLE
bib: use target architecture for buildroot as well

### PR DESCRIPTION
**Pull request by mvo5**
The existing code is trying to fetch two containers when doing a cross-arch image. The buildroot is run in the native architecture and only the target tree is run non-native. However it turns out this is not neccessary as even doing the buildroot in non-native is fast enough (given that it's mostly I/O and all I/O is run naively via syscall translation).

So this commit removes the fetching of the native arch container for the buildroot. Note that in practise this was already the case when passing --target-arch to any container that had native build available. Here due to a bug/misfeature in the resolver library used it already resolved (incorrectly) to the target architecture and nobody noticed because it was good enough (after [0] got fixed).

[0] https://lists.nongnu.org/archive/html/qemu-devel/2024-02/msg03971.html

P.S. I did a benchmark of the centos-stream9,arm64 test on my machine:
- native buildroot: 7:44min
- emulated buildroot: 8:22min

### CODE REVIEW

File: bib/cmd/bootc-image-builder/main.go [hunk 1]
```diff
@@ -170,28 +170,22 @@ func makeManifest(c *ManifestConfig, cacheRoot string) (manifest.OSBuildManifest
...
-	resolverNative := container.NewResolver(hostArch)
-	resolverTarget := resolverNative
-	if hostArch != targetArch {
-		resolverTarget = container.NewResolver(targetArch)
+	var resolver *container.Resolver
+	if targetArch != "" && hostArch != targetArch {
+		resolver = container.NewResolver(targetArch)
+	} else {
+		resolver = container.NewResolver(hostArch)
 	}
```
Comment (cgwalters) [inline bib/cmd/bootc-image-builder/main.go:185]: Isn't this logic just equivalent to:

```
	if targetArch != "" {
		resolver = container.NewResolver(targetArch)
	} else {
		resolver = container.NewResolver(hostArch)
	}
```
If `hostArch == targetArch`, then it's the same thing I think.

**Review by cgwalters — APPROVED** [review]
Looks sane to me, one drive by nit

**Review by kingsleyzissou — DISMISSED** [review]

**Comment by kingsleyzissou** [comment]
> P.S. I did a benchmark of the centos-stream9,arm64 test on my machine:
> 
> - native buildroot: 7:44min
> - emulated buildroot: 8:22min

Neat!!

File: bib/cmd/bootc-image-builder/main.go (hunk 1 above)
Comment (mvo5) [reply bib/cmd/bootc-image-builder/main.go]: Thanks, excellent point. I simplfied it accordingly.

**Review by kingsleyzissou — APPROVED** [review]

**Pull request merged**